### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by cemremengu**

### CODE REVIEW

File: test/services/todo.test.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
+'use strict'
+
+const { test } = require('tap')
+const { build, cleandb } = require('../helper')
+
+test('should create an item', async (t) => {
+  const app = build(t)
+  t.teardown(cleandb)
```
Comment (mcollina) [inline test/services/todo.test.js]: It's better to clean the db _before_ each test, not after. I would put in a 
beforeEach in tap.

**Review by mcollina — COMMENTED** [review]
Good work! Can you add travis as well?

File: test/services/todo.test.js (hunk 1 above)
Comment (cemremengu) [reply test/services/todo.test.js]: Thanks will do. For travis, I will first transfer the repo to org first then add it (I remember it was painful to set it up when transferred)

File: test/plugins/timestamp.test.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+const { test } = require('tap')
+const Fastify = require('fastify')
+const timestamp = require('../../plugins/timestamp')
+
+test('support works standalone', async (t) => {
+  const fastify = Fastify()
+  fastify.register(timestamp)
+
+  t.plan(1)
```
Comment (mcollina) [inline test/plugins/timestamp.test.js]: t.plan is mostly useless with an async function as a test

File: services/todo.js [hunk 3]
```diff
@@ -4,15 +4,19 @@ const schemas = require('../schemas/todo')
...
     return this.mongo.db
       .collection('todo')
       .find()
       .toArray()
   })
 
+  fastify.get('/:name', function (request, reply) {
+    return this.mongo.db
+      .collection('todo')
+      .findOne({ name: request.params.name })
```
Comment (mcollina) [inline services/todo.js:20]: can you add a test if this returns nothing? Do you get a proper 404?

File: test/services/todo.test.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+  const res = await app.inject({
+    url: '/api/todo/error'
+  })
+
+  const payload = JSON.parse(res.payload)
+
+  t.deepEquals(payload, {
+    error: 'Internal Server Error',
+    message: 'boom',
+    statusCode: 500
```
Comment (mcollina) [inline test/services/todo.test.js]: a 500 is never a good thing, it’s a dev error. What’s your intent here?

Comment (cemremengu) [reply test/services/todo.test.js]: I am actually trying to showcase how fastify handles/wraps the thrown error. Do you think this is necessary or maybe a better demonstration?

Comment (cemremengu) [reply test/services/todo.test.js]: I am just removing it, will explain in the write-up/description

File: test/helper.js [hunk 5]
```diff
@@ -28,7 +34,18 @@ function build (t) {
...
 
+async function cleandb (t) {
+  try {
+    const client = await MongoClient.connect(mongo, { w: 1, useNewUrlParser: true })
+    await clean(client.db('todo-test'))
+    await client.close()
+  } catch (err) {
+    t.fail(err)
+  }
+}
```
Comment (mcollina) [inline test/helper.js]: I would recommend something like:

```js
use strict'

// This file contains code that we reuse
// between our tests.

const Fastify = require('fastify')
const fp = require('fastify-plugin')
const App = require('../app')

const clean = require('mongo-clean')
const { MongoClient } = require('mongodb')
const { beforeEach, tearDown, test } = require('tap')
const url = 'mongodb://localhost:27017'
const database = 'tests'

let client

beforeEach(async function () {
  if (!client) {
    client = await MongoClient.connect(url, {
      w: 1,
      useNewUrlParser: true
    })
  }
  await clean(client.db(database))
})

tearDown(async function () {
  if (client) {
    await client.close()
    client = null
  }
})

// Fill in this config with all the configurations
// needed for testing the application
function config () {
  return {
    auth: {
      secret: 'averyverylongsecret'
    },
    mongodb: {
      client,
      database
    }
  }
}

// automatically build and tear down our instance
function build (t) {
  const app = Fastify()

  // fastify-plugin ensures that all decorators
  // are exposed for testing purposes, this is
  // different from the production setup
  app.register(fp(App), config())

  // tear down our app after we are done
  t.tearDown(app.close.bind(app))

  return app
}

module.exports = { build }
```

**Review by mcollina — APPROVED** [review]
LGTM

**Pull request merged**